EjercicioClases/Model/Product.pm

```diff
@@ -32,30 +32,155 @@ sub getSku {
     return $self->{sku};
 }
 
+sub getName {
+    my( $self ) = @_;
+    return $self->{name};
+}
+
+sub getSellerCode {
+    my( $self ) = @_;
+    return $self->{sellerCode};
+}
+
+sub getStock {
+    my( $self ) = @_;
+    return $self->{isAvailable};
+}
+
 sub hasStock {
     my( $self ) = @_;
     return $self->{isAvailable};
 }
 
 sub updateStock {
-   my ( $self, $status ) = @_;
+   my ( $self, $status, $sku ) = @_;
    $self->{ isAvailable } = $status;
+
+   my $dbConnectionString="DBI:mysql:database=PERL_STORE;host=172.17.0.2;mysql_local_infile1";
+   my $dbLogin="root";
+   my $dbPassword="1234";
+   my $dbh;
+   eval{
+      $dbh = DBI->connect($dbConnectionString, $dbLogin, $dbPassword);
+    };
+    if($@){
+      print "Error conectando a db : $@"
+    }
+   my $sth = $dbh->prepare(
+       "UPDATE product SET stock = ? WHERE sku = ?"
+   );
+   $sth->execute($status, $sku) or die $DBI::errstr;
+   $sth->finish();
+   $dbh->disconnect();
    return $self->{isAvailable};
 }
 
 sub getProductBySku{
     my ( $class, $sku ) = @_;
-    return Controller::DatabaseController->retrieveProductBySku($sku);
+
+    my $dbConnectionString="DBI:mysql:database=PERL_STORE;host=172.17.0.2;mysql_local_infile1";
+    my $dbLogin="root";
+    my $dbPassword="1234";
+    my $dbh;
+    eval{
+      $dbh = DBI->connect($dbConnectionString, $dbLogin, $dbPassword);
+    };
+    if($@){
+      print "Error conectando a db : $@"
+    }
+
+    my $sth = $dbh->prepare(
+        "SELECT sku, name, seller_code, stock FROM product WHERE sku = ?"
```

Comment:
If you're expecting to grab just 1 row from the DB table, you may want to add the `LIMIT` statement to your query.